### ISSUE TITLE
content filter/regex

### ISSUE THREAD

**Issue by peteringemann** [poppastring/dasblog-core#361]
dasBlog use to have content filter applied by regex replacement. 
A new addition to bringing content filter back would be to actually apply the filter when search engines visit for indexing. At least the old dasBlog allowed google to index raw content instead of content filtered content.
This is useful if you use content filter to anonymous names etc and prevent google from indexing such.

**Comment by poppastring** [comment]
@peteringemann  I am not sure I understand what this does to be honest. Could you provide some additional info or references or an example of what content is getting changed and how?

**Comment by peteringemann** [comment]
@poppastring The Content Filter was a setting that bascially filtered all posts displayed on the front page. It's a list of entries where every entry was a key-value string. The key being what to search for and the value being the replacement. The Content Filter supported basic replacement or regex replacement. 
Say I created a post with the phrase "The quick brown fox jumps over the lazy dog.". Then I created an entry with key 'brown' and the replacement value 'red', everything that matched 'brown' would be replaced with 'red'. I could also use regex for such.

I used this feature when writing about my friends, and then, to keep them anonymous, I created a content filter entry replacing their full name with a John-Doe-name. The problem was 1) that the Content Filter wasn't applied when Googled trawled the page so Google would index their real name, and 2) the internal search functionality had the same flaw as with Google: it searched the day-entry-xml instead of running the search mechanism through the Content Filter and then search the output. 

At least it would be nice as a setting to set if searching (both internal search or trawlers like Google) should search before or after Content Filter was applied.
 
depicted: the old content filter from dasBlog.
![contentfilter](https://user-images.githubusercontent.com/46843486/73133993-1948d480-4031-11ea-8044-c505d8c0d88c.png)
